src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+      'SpecialismId' => $category,
+      'ProjectMetadata' => [
+        ['MetadataKey' => 'CMS User Name', 'MetadataValue' => \Drupal::currentUser()->getDisplayName()],
+        ['MetadataKey' => 'CMS User Email', 'MetadataValue' => \Drupal::currentUser()->getEmail()],
+        ['MetadataKey' => 'Response Service Base URL', 'MetadataValue' => $_SERVER['HTTP_HOST']],
+        ['MetadataKey' => 'Response Service Path', 'MetadataValue' => '/' . $url->getInternalPath()],
+      ],
+    ];
+    if ($review) {
+      $params['ProjectMetadata'][] = [
+        'MetadataKey' => 'Workflow Options',
+        'MetadataValue' => 'Localize and Review',
+      ];
+    }
+    else {
+      $params['ProjectMetadata'][] = [
+        'MetadataKey' => 'Workflow Options',
+        'MetadataValue' => 'Localize Only',
+      ];
+    }
+
+    return $this->request('project', 'POST', $params);
+  }
+
+  /**
+   * Send the files to Thebigword.
+   *
+   * @param \Drupal\tmgmt\JobItemInterface $job_item
+   *   The Job.
+   */
+  private function sendFiles(JobItemInterface $job_item) {
+    $cache = \Drupal::cache('data');
+    /** @var \Drupal\tmgmt_file\Format\FormatInterface $xliff_converter */
+    $xliff_converter = \Drupal::service('plugin.manager.tmgmt_file.format')->createInstance('xlf');
+    $resource_uuids = array();
+
+    $job_item_id = $job_item->id();
+    $target_language = $job_item->getJob()->getRemoteTargetLanguage();
+    $cid = "tmgmt_thebigword:resource_id:$job_item_id:$target_language";
+    if ($cached = $cache->get($cid)) {
+      $file_id = $cached->data;
+    }
+    else {
+      $conditions = array('tjiid' => array('value' => $job_item_id));
+      $xliff = $xliff_converter->export($job_item->getJob(), $conditions);
+      $name = "JobID_{$job_item->getJob()->id()}_JobItemID_{$job_item_id}_{$job_item->getJob()->getSourceLangcode()}_{$target_language}";
+
+      $file_id = $this->uploadFileResource($xliff, $job_item->getJob(), $job_item_id, $name);
+      $this->sendPreviewUrl($job_item, $file_id, FALSE);
+      $cache->set($cid, $file_id, Cache::PERMANENT, $job_item->getCacheTags());
+    }
+    $resource_uuids[$job_item_id] = $file_id;
+  }
+
+  /**
+   * Creates a file resource at Thebigword.
+   *
+   * @param string $xliff
+   *   .XLIFF string to be translated. It is send as a file.
+   * @param \Drupal\tmgmt\JobInterface $job
+   *   The Job.
+   * @param int $tjiid
+   *   The JobItem id.
+   * @param string $name
+   *   File name of the .XLIFF file.
+   *
+   * @return string Thebigword uuid of the resource.
+   *   Thebigword uuid of the resource.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   */
+  public function uploadFileResource($xliff, JobInterface $job, $tjiid, $name) {
+    $remote_mappings = RemoteMapping::loadByLocalData($job->id(), $tjiid, 'tmgmt_thebigword');
+    $remote_mapping = reset($remote_mappings);
+    $project_id = $remote_mapping->getRemoteIdentifier2();
```

Comment:
relying on the mapping so deep in the API doesn't really seem like a great design to me. I'd prefer passing that information around explicitly.